modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -477,6 +479,13 @@ public SubmosaicProducerFactory getGranuleCollectorsFactory() {
 
     private ReferencedEnvelope mosaicBBox;
 
+    /**
+     * With Heterogeneous CRS, all the geometries in the catalog are expressed in a single CRS, even
+     * if the granules have a different native CRS. This variable is the mosaicBBOX to be used in
```

Comment:
Euh? :-D

Reply:
Whenever CRS you ask in output, the geometries in the catalog are stored into a unique CRS. So the query on the DB need to be made on that CRS even if the requested mosaic BBOX is using a different CRS. I'm going to clarify a bit this comment :)